### PR TITLE
Implement course announcement settings

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/views/course/admin/announcement_settings/index.html.slim [hunk 1]
```diff
@@ -0,0 +1,4 @@
+= simple_form_for @settings, url: course_admin_announcement_path(current_course) do |f|
```
Comment (lowjoel) [inline app/views/course/admin/announcement_settings/index.html.slim]: `simple_form_for()`

File: app/controllers/course/admin/announcement_settings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Course::Admin::AnnouncementSettingsController < Course::Admin::Controller
+  before_action :load_settings, only: [:index, :update]
+
+  def index #:nodoc:
```
Comment (lowjoel) [inline app/controllers/course/admin/announcement_settings_controller.rb]: I think this should be edit.

File: config/routes.rb [hunk 3]
```diff
@@ -90,6 +90,8 @@
 
         get 'components' => 'admin#components'
```
Comment (lowjoel) [inline config/routes.rb]: And maybe this should be extracted out.

Comment (allenwq) [inline config/routes.rb]: Yah, It should, still on my todo list.

File: app/controllers/course/admin/announcement_settings_controller.rb (hunk 2 above)
Comment (allenwq) [inline app/controllers/course/admin/announcement_settings_controller.rb]: :+1:  thanks !

File: app/views/course/admin/announcement_settings/index.html.slim (hunk 1 above)
Comment (allenwq) [inline app/views/course/admin/announcement_settings/index.html.slim]: Then all exiting code need change ?  I actually prefer no brackets, and this also follows the style in official `simple_form` guide.  I know you have concern about DSL, but..

Comment (lowjoel) [inline app/views/course/admin/announcement_settings/index.html.slim]: More without brackets, so should remove the brackets then.

Feels inconsistent to me, but okay.

Comment (allenwq) [inline app/views/course/admin/announcement_settings/index.html.slim]: Thanks ！

**Comment by allenwq** [comment]
@lowjoel  do you have more comments about this ?

File: config/routes.rb [hunk 4]
```diff
@@ -90,6 +90,8 @@
...
         patch 'components' => 'admin#update_components'
+        get 'announcement' => 'announcement_settings#edit'
```
Comment (lowjoel) [inline config/routes.rb]: since component*s*, announcement*s*

Comment (lowjoel) [inline config/routes.rb]: i'll merge after that.

Comment (allenwq) [inline config/routes.rb]: Sure

File: app/models/course/announcement/settings.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
+class Course::Announcement::Settings
```
Comment (lowjoel) [inline app/models/course/announcement/settings.rb]: Should this be Course::Announcement::Settings or Course::AnnouncementSettings?

Comment (allenwq) [inline app/models/course/announcement/settings.rb]: I think should be `Course::Announcement::Settings` ? similar to `Course::Settings`

Comment (lowjoel) [inline app/models/course/announcement/settings.rb]: Opening another namespace seems extraneous to me. Your model and controller should correspond.

Comment (allenwq) [inline app/models/course/announcement/settings.rb]: Sure.

**Comment by allenwq** [comment]
@lowjoel done.

**Pull request merged**